src/puppetlabs/trapperkeeper/services/metrics/metrics_service.clj

```diff
@@ -7,11 +7,40 @@
 (trapperkeeper/defservice metrics-service
   metrics/MetricsService
   [[:ConfigService get-in-config]]
+
   (init [this context]
-    (core/initialize (get-in-config [:metrics] {})))
+    {:registries
+     (atom {::default-registry
+            (core/initialize (get-in-config [:metrics] {})
+                             nil)})})
+
   (stop [this context]
-    (core/stop (tk-services/service-context this)))
+    (let [{:keys [registries] :as ctx} (tk-services/service-context this)]
+      (doseq [[_ metrics-reg] @registries]
```

Comment:
Is there a race condition here if I call ```core/get-or-initialize!``` at the same time? Or does TK prevent this?